src/main/java/com/amazon/opendistroforelasticsearch/security/configuration/OpenDistroSecurityIndexSearcherWrapper.java

```diff
@@ -87,7 +93,12 @@ public final DirectoryReader apply(DirectoryReader reader) throws IOException {
         if (isSecurityIndexRequest() && !isAdminAuthenticatedOrInternalRequest()) {
```

Comment:
should we add isSystemIndexRequest() check here?

Reply:
Had a offline discussion and decided to keep it current way.

Reply:
IMO, the entire class needs to be modified. I'll do this in a follow up PR.